src/services/refactors/extractType.ts

```diff
@@ -260,14 +264,14 @@ function rangeContainsSkipTrivia(r1: TextRange, node: TextRange, file: SourceFil
     return rangeContainsStartEnd(r1, skipTrivia(file.text, node.pos), node.end);
 }
 
-function collectTypeParameters(checker: TypeChecker, selection: TypeNode | TypeNode[], enclosingNode: Node, file: SourceFile): TypeParameterDeclaration[] | undefined {
+function collectTypeParameters(checker: TypeChecker, selection: TypeNode | TypeNode[], enclosingNode: Node, file: SourceFile): { typeParameters: TypeParameterDeclaration[] | undefined; affectedTextRange: TextRange | undefined; } {
     const result: TypeParameterDeclaration[] = [];
     const selectionArray = toArray(selection);
     const selectionRange = { pos: selectionArray[0].pos, end: selectionArray[selectionArray.length - 1].end };
     for (const t of selectionArray) {
-        if (visitor(t)) return undefined;
+        if (visitor(t)) return { typeParameters: undefined, affectedTextRange: undefined };
     }
-    return result;
+    return { typeParameters: result, affectedTextRange: selectionRange };

```

Comment:
Small nit: the beginning of `selectionRange` includes all the leading whitespace before the first type parameter, which may look weird to the user now that it will be visible. Not a huge deal, but we should be somewhat careful that ranges that previously existed only for programmatic use look reasonable when turned into a UI element.